app/controllers/common/ChooseTaxYearController.scala

```diff
@@ -40,21 +42,24 @@ class ChooseTaxYearController @Inject()(val controllerComponents: MessagesContro
 
   def onPageLoad(waypoints: Waypoints, eventType: EventType, index: Index): Action[AnyContent] =
     (identify andThen getData(eventType)) { implicit request =>
-    val form = formProvider(eventType)
-    val preparedForm = request.userAnswers.flatMap(_.get(common.ChooseTaxYearPage(eventType, index))).fold(form)(form.fill)
-    Ok(view(preparedForm, waypoints, eventType, index))
+      val taxYearChosen = getTaxYearFromOption(request.userAnswers)
+    val form = formProvider(eventType, taxYearChosen)
+      val rdsTaxYear = ChooseTaxYear.reads(ChooseTaxYear.enumerable(taxYearChosen))
+    val preparedForm = request.userAnswers.flatMap(_.get(common.ChooseTaxYearPage(eventType, index))(rdsTaxYear)).fold(form)(form.fill)
+    Ok(view(preparedForm, waypoints, eventType, index, taxYearChosen))
   }
 
   def onSubmit(waypoints: Waypoints, eventType: EventType, index: Index): Action[AnyContent] =
     (identify andThen getData(eventType)).async {
     implicit request =>
-      val form = formProvider(eventType)
+      val taxYearChosen = getTaxYearFromOption(request.userAnswers)
+      val form = formProvider(eventType, taxYearChosen)
       form.bindFromRequest().fold(
         formWithErrors =>
-          Future.successful(BadRequest(view(formWithErrors, waypoints, eventType, index))),
+          Future.successful(BadRequest(view(formWithErrors, waypoints, eventType, index, getTaxYearFromOption(request.userAnswers)))),
```

Comment:
This getTaxYearFromOption method (and the getTaxYear method) I think don't belong on the datehelper. I would put them on the UserAnswers class and then add to controllers requireData so you don't have to worry about the option